native/packages/react-native-bpk-component-picker/src/BpkPickerMenu.android.js

```diff
@@ -27,64 +33,109 @@ import {
   lineHeightBase,
   spacingBase,
 } from 'bpk-tokens/tokens/base.react.native';
+import BpkPickerItem from './BpkPickerItem';
 
-const rowsToDisplay = 6;
-
-const maxListHeight =
-  // eslint-disable-next-line no-mixed-operators
-  (2 * spacingBase + lineHeightBase) * rowsToDisplay;
+const MAX_ROWS_TO_DISPLAY = 6;
 
 const styles = StyleSheet.create({
   overlay: {
-    flex: 1,
+    left: 0,
+    height: '100%',
+    position: 'absolute',
+    top: 0,
+    width: '100%',
     backgroundColor: setOpacity(colorGray900, 0.8),
-    justifyContent: 'center',
   },
   list: {
     backgroundColor: colorWhite,
     borderRadius: borderRadiusSm,
-    maxHeight: maxListHeight,
-    margin: spacingBase,
-    flex: 0,
+    alignSelf: 'center',
+    top: '50%',
+    minWidth: '50%',
   },
 });
 
-const PickerMenu = props => {
+const BpkPickerMenu = props => {
   const { visible, children, onValueChange, onClose, selectedValue } = props;
-  const pickerItems = React.Children.map(children, (child, index) =>
-    React.cloneElement(child, {
-      selected: selectedValue && selectedValue === child.props.value,
-      onPress: value => {
-        onValueChange(value, index);
-        onClose();
-      },
-    }),
-  );
+
+  let initialScrollIndex = 0;
+  const pickerItems = React.Children.map(children, (child, index) => {
+    const { label, value } = child.props;
+
+    let selected = false;
+    if (selectedValue && selectedValue === value) {
+      selected = true;
+
+      // If selected item will be off-screen, set initialScrollIndex.
+      if (index >= MAX_ROWS_TO_DISPLAY) {
+        initialScrollIndex = index - 2;
+      }
+    }
+    return { index, label, value, selected };
+  });
+
+  const rowsToDisplay =
+    pickerItems.length > MAX_ROWS_TO_DISPLAY
+      ? MAX_ROWS_TO_DISPLAY
+      : pickerItems.length;
+  const heightOfOneItem = spacingBase * 2 + lineHeightBase;
```

Comment:
To be refactored as this is currently fragile, will break if `BpkPickerItem` style changes.